OKTA-434102 Add ability to use OIDC access tokens. #508 (#517)
okta/okta-sdk-dotnet: README.md

var client = new OktaClient(clientConfiguration);
```

> It is possible to use previously requested access token for an authentication. For that set `AuthorizationMode` configuration property to `AuthorizationMode.OAuthAccessToken` and `OAuthAccessToken`  to the token string. It is also possible to provide a token renewing function, SDK will call the function when access token expires.

Reviewer: ```suggestion
It is possible to use previously requested access token for an authentication. For that, set the `AuthorizationMode` configuration property to `AuthorizationMode.OAuthAccessToken` and the `OAuthAccessToken`  property to the token string. It is also possible to provide a token renewing function, SDK will call the function when access token expires.
```

```csharp
    var client = new OktaClient(new OktaClientConfiguration
        {
            ClientId = "{{clientId}}",
            AuthorizationMode = AuthorizationMode.BearerToken,
            BearerToken = "{{preRequestedAccessToken}}",
        },

>         oauthTokenRenewer: async () =>

Reviewer 2: SHouldn't this param be an IOAuthTokenProvider implementation instead?

Author: Yes! I forgot to change  the sample after implementation was changed

var client = new OktaClient(clientConfiguration);
```

> It is possible to use previously requested access token for an authentication. For that set `OktaClientConfiguration.AuthorizationMode` configuration property to `AuthorizationMode.BearerToken` and `OktaClientConfiguration.BearerToken` to the token string. 

Reviewer 2: ```suggestion
It is possible to use an access token you retrieved outside of the SDK for authentication. For that, set `OktaClientConfiguration.AuthorizationMode` configuration property to `AuthorizationMode.BearerToken` and `OktaClientConfiguration.BearerToken` to the token string. 
```

> To provide already generated token, it's also possible to create a custom class implementing `IOAuthTokenProvider` interface and pass the instance of the class to OktaClient constructor. 

Reviewer 2: In addition to passing the token via configuration, you can inject your own implementation of the `IOAuthTokenProvider` interface via the OktaClient constructor. This strategy is useful when you want to refresh a token that has been expired.

> You can provide a value for `OktaClientConfiguration.BearerToken` option together with a custom `IOAuthTokenProvider` implementation. In this case value from `OktaClientConfiguration.BearerToken` will be used first and if request fails (for example when the token expires) the custom token provider will be called.

Reviewer 2: ```suggestion
You can provide a value for `OktaClientConfiguration.BearerToken` option along with a custom `IOAuthTokenProvider` implementation. In this case, the SDK will try to use the  `OktaClientConfiguration.BearerToken` first and if the request fails (for example when the token expires) then the SDK will retry with the custom token provider.
```
